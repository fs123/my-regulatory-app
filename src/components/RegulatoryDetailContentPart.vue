<template>
  <div>
    <div v-for="element in contentElements" :key="element.text">
      <v-row>
        <v-col v-if="element.type === 'HEADING'" cols="8" md="12">
          <v-card>
            <h1>{{element.text}}</h1>
          </v-card>
        </v-col>
        <v-col v-if="element.type === 'TEXT'" cols="8" md="8">
          <v-card>
            <span class="nodeClass">{{element.text}}</span>
          </v-card>
        </v-col>
        <v-col v-if="element.type === 'TEXT'" cols="4" md="4">
          <v-card>
            <span class="nodeClass">SECOND {{element.text.substring(0, 150)}}</span>
          </v-card>
        </v-col>
      </v-row>
      <RegulatoryDetailContentPart
        v-if="element.type === 'HEADING'"
        :contentElements="element.content"
      />
    </div>
  </div>
</template>

<script lang="ts">
export default {
  name: "RegulatoryDetailContentPart",
  props: ["contentElements"],
};
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped>
h3 {
  margin: 40px 0 0;
}
.nodeClass {
  white-space: pre-line;
}

li {
  text-align: left;
}
a {
  color: #42b983;
}
</style>
