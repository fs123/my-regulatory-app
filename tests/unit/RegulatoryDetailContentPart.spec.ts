import Vue from 'vue'
import Vuetify from 'vuetify'
import { shallowMount, mount } from '@vue/test-utils';
import RegulatoryDetailContentPart from '@/components/RegulatoryDetailContentPart.vue';

const TWO_ELEMENTS = [
  {
    "type": "TEXT",
    "text": "Element 1"
  },
  {
    "type": "TEXT",
    "text": "Element 2"
  }
];

describe('Test Regulatory Detail Content Part', () => {
  it('renders no entry with empty contentElements', () => {
    const wrapper = shallowMount(RegulatoryDetailContentPart, {
      propsData: { contentElements: [] },
    });
    
    expect(wrapper.element.tagName).toBe('DIV');
    expect(wrapper.element.innerHTML).toBe('');
  });

  it('renders two text elements', () => {
    const wrapper = mount(RegulatoryDetailContentPart, {
      propsData: { contentElements: TWO_ELEMENTS },
      
    });
    
    expect(wrapper.element.tagName).toBe('DIV');
    expect(wrapper.html()).toMatchSnapshot();
  });
});
