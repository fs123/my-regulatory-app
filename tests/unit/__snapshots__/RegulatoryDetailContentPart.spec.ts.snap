// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Regulatory Detail Content Part renders two text elements 1`] = `
<div>
  <div>
    <div class="row">
      <!---->
      <div class="col-md-8 col-8">
        <div class="v-card v-sheet theme--light"><span>Element 1</span></div>
      </div>
      <div class="col-md-4 col-4">
        <div class="v-card v-sheet theme--light"><span>SECOND Element 1</span></div>
      </div>
    </div>
    <!---->
  </div>
  <div>
    <div class="row">
      <!---->
      <div class="col-md-8 col-8">
        <div class="v-card v-sheet theme--light"><span>Element 2</span></div>
      </div>
      <div class="col-md-4 col-4">
        <div class="v-card v-sheet theme--light"><span>SECOND Element 2</span></div>
      </div>
    </div>
    <!---->
  </div>
</div>
`;
